src/libsync/pushnotifications.cpp

```diff
@@ -194,4 +205,76 @@ void PushNotifications::handleNotifyActivity()
     qCInfo(lcPushNotifications) << "Push activity arrived";
     emit activitiesChanged(_account);
 }
+void PushNotifications::onWebSocketPongReceived(quint64 /*elapsedTime*/, const QByteArray &payload)
+{
+    handleTimeoutPong(payload);
+}
+
+void PushNotifications::handleTimeoutPong(const QByteArray &payload)
+{
+    // We are not interested in different pongs from the server
+    const auto expectedPingPayload = getTimeoutPingPayload();
+    if (payload != expectedPingPayload) {
+        return;
+    }
+
+    qCDebug(lcPushNotifications) << "Pong received in time";
+
+
+    _pingTimedOutTimer.stop();
+    _timeoutPongReceivedFromWebSocketServer = true;
+    startPingTimeoutTimer();
+}
+
+void PushNotifications::startPingTimeoutTimer()
+{
+    _pingTimeoutTimer.setInterval(_pingTimeoutInterval);
+    _pingTimeoutTimer.setSingleShot(true);
+    connect(&_pingTimeoutTimer, &QTimer::timeout, this, &PushNotifications::pingWebSocketServer, Qt::UniqueConnection);
+    _pingTimeoutTimer.start();
+}
+
+void PushNotifications::startPingTimedOutTimer()
```

Comment:
These call for better names: "pingTimeoutTimer" vs "pingTimedOutTimer" are waaaay too close.

One drives the ping right? so it's just the pingTimer I guess.
The other ones is to catch the case when the pong never came back? this would be a pingTimeoutTimer, or pingExpiredTimer or such

Reply:
Initially the name was  `pingTimer`. I thought this name would be not specific enough. But yeah I you're right, the names are too close.

Reply:
@FlexW @er-vin Or, even better, since we are waiting for the server to **pong**, why not name it something like **startWaitPongFromServer**, or **startPongTimedoutTimer** or such?

Reply:
@allexzander That would be a mistake because this timer gets only executed if the ping **timedOut**. There is a other method that handles the pong response.